src/Umbraco.Core/Composing/TypeFinder.cs

```diff
@@ -210,53 +210,55 @@ internal static HashSet<Assembly> GetAllAssemblies()
         /// NOTE the comma vs period... comma delimits the name in an Assembly FullName property so if it ends with comma then its an exact name match
         /// NOTE this means that "foo." will NOT exclude "foo.dll" but only "foo.*.dll"
         /// </remarks>
-        internal static readonly string[] KnownAssemblyExclusionFilter = new[]
-                {
-                    "mscorlib,",
-                    "System.",
-                    "Antlr3.",
-                    "Autofac.",
-                    "Autofac,",
-                    "Castle.",
-                    "ClientDependency.",
-                    "DataAnnotationsExtensions.",
-                    "DataAnnotationsExtensions,",
-                    "Dynamic,",
-                    "HtmlDiff,",
-                    "Iesi.Collections,",
-                    "Microsoft.",
-                    "Newtonsoft.",
-                    "NHibernate.",
-                    "NHibernate,",
-                    "NuGet.",
-                    "RouteDebugger,",
-                    "SqlCE4Umbraco,",
-                    "umbraco.datalayer,",
-                    "umbraco.interfaces,",
-                    //"umbraco.providers,",
-                    //"Umbraco.Web.UI,",
-                    "umbraco.webservices",
-                    "Lucene.",
-                    "Examine,",
-                    "Examine.",
-                    "ServiceStack.",
-                    "MySql.",
-                    "HtmlAgilityPack.",
-                    "TidyNet.",
-                    "ICSharpCode.",
-                    "CookComputing.",
-                    "AutoMapper,",
-                    "AutoMapper.",
-                    "AzureDirectory,",
-                    "itextsharp,",
-                    "UrlRewritingNet.",
-                    "HtmlAgilityPack,",
-                    "MiniProfiler,",
-                    "Moq,",
-                    "nunit.framework,",
-                    "TidyNet,",
-                    "WebDriver,"
-                };
+        internal static readonly string[] KnownAssemblyExclusionFilter = {
```

Comment:
This is an arms race. You'd be much better off registering your own assemblies and creating a simple means for consumers to register their own assemblies for discovery.

Reply:
This list is old and mostly arbitrary - i mentioned on the other thread, we don't scan all assemblies, we only scan assemblies that have references to our own assemblies for the types being scanned, similar to how .net core scans assemblies for controllers.